src/main/java/hudson/plugins/jira/Updater.java

```diff
@@ -321,6 +281,79 @@ private String getScmComments(boolean wikiStyle,
     }
 
+    protected String createScmChangeEntryDescription(Run<?, ?> run, Entry change, boolean wikiStyle,
+            boolean recordScmChanges) {
+        StringBuilder description = new StringBuilder();
+        RepositoryBrowser repoBrowser = getRepositoryBrowser(run);
+        JiraSite site = JiraSite.get(run.getParent());
+
+        description.append(change.getMsg());
+        String revision = getRevision(change);
+        if (revision != null) {
+            URL url = null;
+            if (repoBrowser != null) {
+                try {
+                    url = repoBrowser.getChangeSetLink(change);
+                } catch (IOException e) {
+                    LOGGER.log(Level.WARNING, "Failed to calculate SCM repository browser link", e);
+                }
+            }
+            description.append(" (");
+            String uid = change.getAuthor().getId();
+            if (StringUtils.isNotBlank(uid)) {
+                description.append(uid).append(": ");
+            }
+            long timestamp = change.getTimestamp();
+            if (timestamp > 0) {
+                appendChangeTimestampToDescription(description, site, timestamp);
+                description.append(" ");
+            }
+            if (url != null && StringUtils.isNotBlank(url.toExternalForm())) {
+                if (wikiStyle) {
+                    description.append("[").append(revision).append("|");
+                    description.append(url.toExternalForm()).append("]");
+                } else {
+                    description.append("[").append(url.toExternalForm()).append("]");
+                }
+            } else {
+                description.append("rev ").append(revision);
+            }
+            description.append(")");
+        }
+        description.append("\n");
+        if (recordScmChanges) {
+            // see http://issues.jenkins-ci.org/browse/JENKINS-2508
+            // added additional try .. catch; getAffectedFiles is not supported
+            // by all SCM implementations
+            try {
+                for (AffectedFile affectedFile : change.getAffectedFiles()) {
+                    description.append("* (").append(affectedFile.getEditType()).append(") ")
+                            .append(affectedFile.getPath()).append("\n");
+                }
+            } catch (UnsupportedOperationException e) {
+                LOGGER.warning("Unsupported SCM operation 'getAffectedFiles'. Fall back to getAffectedPaths.");
+                for (String affectedPath : change.getAffectedPaths()) {
+                    description.append("* ").append(affectedPath).append("\n");
+                }
+            }
+            description.append("\n");
+        }
+        return description.toString();
+    }
+
+    protected void appendChangeTimestampToDescription(StringBuilder description, JiraSite site, long timestamp) {
+        DateFormat df = null;
+        if (site.getDateTimePattern() != null) {
```

Comment:
I like this feature, but I think we need to make it optional, either preserve the former behavior when the format field is empty, or with a checkbox.